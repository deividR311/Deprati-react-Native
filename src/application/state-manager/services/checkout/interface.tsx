import { DeliveryOptionsResponse } from '../../../../infrastucture/apis/delivery-address';
import { Cart } from '../../../../infrastucture/apis/shopping-cart';
import { CheckoutSteps } from '../../../../presentation/navigation/checkout/interfaces';

export interface CheckoutStateHook {
  onContinueButtonTrigger?: CheckoutSteps;
  onPressContinueButton: (screenId: CheckoutSteps) => void;
}

export interface CheckoutState {
  screenId?: CheckoutSteps;
  previousScreen?: CheckoutSteps;
  stackSummary?: any[];
  showLoadingScreen?: boolean;
  onClickedContinueButton: boolean;
  cart: Partial<Cart>;
  totalPriceBeforeSelectDeliveryMode?: string;
  needRefreshCart?: boolean;
  deliveryOptions?: DeliveryOptionsResponse;
  isGiftCardCart: boolean;
  creditInfo?: {
    saveForFuturePay: boolean;
    maskedPhoneNumber: string;
    maskedNumberAccount: string;
    maskedEmail: string;
    directCreditCodeCustomer?: string;
    directCreditCodeAdditional?: string;
  };
  firstTimeSummary?: boolean;
}

export const INITIAL_STATE: CheckoutState = {
  screenId: undefined,
  previousScreen: undefined,
  stackSummary: undefined,
  onClickedContinueButton: false,
  showLoadingScreen: false,
  needRefreshCart: false,
  cart: {},
  deliveryOptions: undefined,
  totalPriceBeforeSelectDeliveryMode: undefined,
  isGiftCardCart: false,
  creditInfo: undefined,
  firstTimeSummary: undefined
};
