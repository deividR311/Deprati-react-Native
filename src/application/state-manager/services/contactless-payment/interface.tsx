import { ContactLessPaymentSteps } from '../../../../presentation/navigation/contactless-payment';

export enum BottomSheetType {
  IDENTITY_VALIDATION = 'IDENTITY_VALIDATION',
  SUCCESS_BUY = 'SUCCESS_BUY'
}

export interface ShowBottomSheetAction {
  bottomSheet: BottomSheetType;
  show: boolean;
  text?: string;
  ticketId?: string;
}

export interface ContactlessPaymentStateHook {
  TicketNumberContactLessPayment?: string;
  showIdentityValidation(show: boolean): void;
  identityValidationIsOpen: boolean;
  showSuccessBuy(show: boolean, text?: string, ticketId?: string): void;
  successBuyIsOpen: boolean;
  successBuyText?: string;
  successBuyTicketId?: string;
  doEnableConfirmPurchaseButton(isEnable: boolean, ticketId?: string): void;
  enableConfirmPurchaseButton: boolean;
}

export interface ContactlessPaymentState {
  tabbar: {
    disableNextButton: boolean;
    disablePreviousButton: boolean;
    currentTab: ContactLessPaymentSteps | '';
  };
  bottomSheetState: {
    identityValidationIsOpen: boolean;
    successBuyIsOpen: boolean;
    successBuyText: string;
    successBuyTicketId: string;
  };
  enableConfirmPurchaseButton: boolean;
}

export const INITIAL_STATE: ContactlessPaymentState = {
  tabbar: {
    currentTab: '',
    disableNextButton: false,
    disablePreviousButton: false
  },
  bottomSheetState: {
    identityValidationIsOpen: false,
    successBuyIsOpen: false,
    successBuyText: '',
    successBuyTicketId: ''
  },
  enableConfirmPurchaseButton: false
};
