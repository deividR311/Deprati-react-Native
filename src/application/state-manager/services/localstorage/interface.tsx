import { NotificationListRes } from '../../../../infrastucture/apis/notifications/notifications.interfaces';
import { MessagesAppInterface } from '../../../../presentation/screens/Splash/hooks/messages.interface';
import { INITIAL_STATE as ContactlessInitialState } from '../contactless-payment';
import {
  UserResponse,
  USER_DEFAULT_DATA
} from '../../../../infrastucture/apis/user/user.interface';

export type LocalStorageParams = Partial<LocalStorageType>;

export interface LocalStorageHook {
  localStorageData: LocalStorageType;
  save(params: LocalStorageParams): void;
  remove(keys: LocalStorageKey[]): void;
  readAndStorageInState(): void;
  fullClean(): void;
}

export enum LocalStorageKey {
  // BEHAVIOR
  WasStoreRead = 'WAS_STORE_READ',
  isAppOpenByFirstTime = 'IS_APP_OPEN_BY_FIRST_TIME',
  IpAddress = 'IP_ADDRESS',
  UID = 'DEVICE_ID',
  IsEnableConfirmPurchaseButtonInQRScreen = 'CONTACTLESS_PAYMENT@IS_ENABLE_CONFIRM_PURCHASE_BUTTON',
  TicketNumberContactLessPayment = 'CONTACTLESS_PAYMENT@TICKET_NUMBER',
  AnonymousCartGuid = 'ANONYMOUS_CART_GUID',

  // USER DATA
  User = 'USER_DATA',
  UserEmail = 'USER_EMAIL',
  UserLastEntry = 'USER_LAST_ENTRY',

  // ACCOUNT DATA
  IsAccountAuthenticated = 'IS_ACCOUNT_AUTHENTICATED',
  AccountNumber = 'ACCOUNT_NUMBER',
  AccountDisplayNumber = 'ACCOUNT_DISPLAY_NUMBER',
  AccountAdditionalNumber = 'ACCOUNT_ADDITIONAL_NUMBER',
  AccountPhoneNumber = 'ACCOUNT_DISPLAY_PHONE_NUMBER',
  LastCustomerId = 'LAST_CUSTOMER_ID',

  //Notifications
  Notifications = 'NOTIFICATIONS',

  //Notifications Push FCM
  TokenFCM = 'TOKENFCM',

  // SECURITY
  Token = 'TOKEN',
  RefreshToken = 'REFRESH_TOKEN',
  ExpiresIn = 'EXPERES_IN',
  ExpiresAt = 'EXPERES_AT',
  GetTokenDate = 'GET_TOKEN_DATE',
  IsLogin = 'IS_LOGIN',
  IsSucessFirstLogin = 'IS_SUCCESS_FIRST_LOGIN',

  BiometricPublicKey = 'publicKey-generate',
  MessagesApp = 'MESSAGES_APP',
  LocalStorageKey = 'LocalStorageKey',
  Whatsapp = 'WHATSAPP'
}

export interface LocalStorageType {
  [LocalStorageKey.WasStoreRead]: boolean;
  [LocalStorageKey.isAppOpenByFirstTime]: boolean;
  [LocalStorageKey.UserLastEntry]: string;
  [LocalStorageKey.UserEmail]: string;
  [LocalStorageKey.AccountNumber]: string;
  [LocalStorageKey.AccountDisplayNumber]: string;
  [LocalStorageKey.AccountAdditionalNumber]: string;
  [LocalStorageKey.AccountPhoneNumber]: string;
  [LocalStorageKey.LastCustomerId]: string;
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]: boolean;
  [LocalStorageKey.TicketNumberContactLessPayment]?: string;
  [LocalStorageKey.AnonymousCartGuid]?: string;
  [LocalStorageKey.User]: UserResponse;
  [LocalStorageKey.Token]: string;
  [LocalStorageKey.RefreshToken]: string;
  [LocalStorageKey.ExpiresIn]: number;
  [LocalStorageKey.ExpiresAt]: string;
  [LocalStorageKey.GetTokenDate]: string;
  [LocalStorageKey.IsLogin]: boolean;
  [LocalStorageKey.IsAccountAuthenticated]: boolean;
  [LocalStorageKey.IpAddress]: string;
  [LocalStorageKey.BiometricPublicKey]: string;
  [LocalStorageKey.UID]: string;
  [LocalStorageKey.IsSucessFirstLogin]: boolean;
  [LocalStorageKey.Notifications]: {
    unread: number;
    notifications: NotificationListRes[];
  };
  [LocalStorageKey.MessagesApp]: MessagesAppInterface;
  [LocalStorageKey.TokenFCM]: string;
  [LocalStorageKey.Whatsapp]: {
    message: string;
    phone: string;
  };
}

export const InitialLocalStorage: LocalStorageType = {
  [LocalStorageKey.WasStoreRead]: false,
  [LocalStorageKey.isAppOpenByFirstTime]: true,
  [LocalStorageKey.AccountNumber]: '',
  [LocalStorageKey.AccountDisplayNumber]: '',
  [LocalStorageKey.AccountAdditionalNumber]: '',
  [LocalStorageKey.AccountPhoneNumber]: '',
  [LocalStorageKey.LastCustomerId]: '',
  [LocalStorageKey.User]: USER_DEFAULT_DATA,
  [LocalStorageKey.UserEmail]: '',
  [LocalStorageKey.UserLastEntry]: '',
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]:
    ContactlessInitialState.enableConfirmPurchaseButton,
  [LocalStorageKey.TicketNumberContactLessPayment]: '',
  [LocalStorageKey.AnonymousCartGuid]: '',
  [LocalStorageKey.Token]: '',
  [LocalStorageKey.RefreshToken]: '',
  [LocalStorageKey.ExpiresIn]: 0,
  [LocalStorageKey.ExpiresAt]: '',
  [LocalStorageKey.GetTokenDate]: '',
  [LocalStorageKey.IsLogin]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IpAddress]: '0.0.0.0',
  [LocalStorageKey.BiometricPublicKey]: '',
  [LocalStorageKey.UID]: '',
  [LocalStorageKey.IsSucessFirstLogin]: false,
  [LocalStorageKey.Notifications]: {
    unread: 0,
    notifications: []
  },
  [LocalStorageKey.MessagesApp]: {},
  [LocalStorageKey.TokenFCM]: '',
  [LocalStorageKey.Whatsapp]: {
    phone: '',
    message: ''
  }
};
