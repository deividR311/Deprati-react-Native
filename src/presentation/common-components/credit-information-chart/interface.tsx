export interface CreditInformationChartProps {
  ownerFullname?: string;
  cardNumber?: string;
  affiliateDate?: string;
  totalAmount?: number;
  amountSpent?: number;
  availableAmount?: number;
  numeroTarjetaAdicionalDisplay?: string;
  showAffiliateDate?: boolean;
  showDisplayCard?: boolean;
  displayCartType?: 'small' | 'large';
  invertValues?: boolean;
  constentChart: any;
  wrapperChartInCard?: boolean;
}

export interface DonutChartProps {
  indicatorValue?: number;
  maxIndicatorValue?: number;
  color?: string;
  dialColor?: string;
  radius?: number;
  strokeWidth?: number;
  duration?: number;
  delay?: number;
  invertValues?: boolean;
}
