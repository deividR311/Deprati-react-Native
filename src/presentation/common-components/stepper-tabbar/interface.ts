import {
  BottomTabBarProps,
  BottomTabNavigationOptions
} from '@react-navigation/bottom-tabs';
import { ParamListBase, TabNavigationState } from '@react-navigation/native';

export interface PropsButtonTab {
  arrowIconDirection: 'left' | 'right';
  options?: BottomTabNavigationOptions;
  isFocused?: boolean;
  onPress?: () => void;
  onLongPress?: () => void;
}

export interface StepperTabBarNavigationProps extends BottomTabBarProps {
  title?: string;
  subtitle?: string;
  showMenu?: () => void;
  onBackFromTabbar?(state: TabNavigationState<ParamListBase>): void;
  onNextFromTabbar?(state: TabNavigationState<ParamListBase>): void;
  showButtomsWhenZero?: boolean;
}
