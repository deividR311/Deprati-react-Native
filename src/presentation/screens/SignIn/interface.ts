export interface CommonDataUserSocialNetwork {
  firsname: string | null;
  lastname: string | null;
  email: string | null;
}
