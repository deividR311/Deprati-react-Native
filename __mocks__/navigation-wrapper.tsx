import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
