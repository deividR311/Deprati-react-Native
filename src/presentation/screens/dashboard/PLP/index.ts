export { default as PLPScreen } from './PLPScreen';
export * from './interfaces';
export * from './views';
