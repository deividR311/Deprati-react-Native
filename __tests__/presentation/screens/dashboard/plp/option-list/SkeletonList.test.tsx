import React from 'react';
import { render } from '@testing-library/react-native';
import { IRender } from '../../../../../../__mocks__/IRender.interface';
import { NavigationWrapper } from '../../../../../../__mocks__/navigation-wrapper';
import { SkeletonList } from '../../../../../../src/presentation/screens/dashboard/PLP';

let screenTest: IRender;

describe('Skeleton PLP', () => {
  it('renders correctly in Snapshot Skeleton List', () => {
    screenTest = render(<NavigationWrapper screen={SkeletonList} />);
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
