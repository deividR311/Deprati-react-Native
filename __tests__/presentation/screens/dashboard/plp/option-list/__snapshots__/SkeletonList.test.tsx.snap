// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton PLP renders correctly in Snapshot Skeleton List 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTScrollView
          ItemSeparatorComponent={[Function]}
          ListHeaderComponent={[Function]}
          data={
            Array [
              1,
              2,
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "width": "100%",
            }
          }
          testID="skeleton-list"
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
            >
              <View
                style={
                  Object {
                    "height": 20,
                  }
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginTop": 30,
                    "paddingBottom": 70,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 200,
                          "key": "Image",
                          "marginBottom": 6,
                          "overflow": "hidden",
                          "width": 150,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "alignSelf": "auto",
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 25,
                          "key": "title",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 120,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 25,
                          "key": "price",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 90,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 10,
                          "height": 20,
                          "key": "color",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 10,
                          "height": 20,
                          "key": "star",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 55,
                          "key": "text",
                          "marginLeft": 7,
                          "marginTop": 10,
                          "overflow": "hidden",
                          "width": 120,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 10,
                  }
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginTop": 30,
                    "paddingBottom": 70,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 200,
                          "key": "Image",
                          "marginBottom": 6,
                          "overflow": "hidden",
                          "width": 150,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "alignSelf": "auto",
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 25,
                          "key": "title",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 120,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 25,
                          "key": "price",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 90,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 10,
                          "height": 20,
                          "key": "color",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 10,
                          "height": 20,
                          "key": "star",
                          "marginLeft": 7,
                          "marginTop": 5,
                          "overflow": "hidden",
                          "width": 20,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E1E9EE",
                          "borderRadius": 4,
                          "height": 55,
                          "key": "text",
                          "marginLeft": 7,
                          "marginTop": 10,
                          "overflow": "hidden",
                          "width": 120,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "height": "100%",
                              "position": "absolute",
                              "width": "100%",
                            },
                            Object {
                              "transform": Array [
                                Object {
                                  "translateX": undefined,
                                },
                              ],
                            },
                          ]
                        }
                      >
                        <BVLinearGradient
                          colors={
                            Array [
                              4292995566,
                              4294113532,
                              4292995566,
                            ]
                          }
                          endPoint={
                            Object {
                              "x": 1,
                              "y": 0,
                            }
                          }
                          locations={null}
                          startPoint={
                            Object {
                              "x": 0,
                              "y": 0,
                            }
                          }
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
